codetransformer/transformers/literals.py

```diff
@@ -42,3 +44,56 @@ def visit_STORE_MAP(self, instr):
 
         yield Instruction(ops.STORE_SUBSCR)
         # TOS  = m
+
+
+def _format_constant_docstring(type_):
+    return dedent(
+        """
+        Transformer that applies a callable to each {type_} constant in the
+        transformed code object
+
+        Parameters
+        ----------
+        f : callable
+            A callable to be applied to {type_} literals.
+        """
+    ).format(type_=type_.__name__)
+
+
+class _ConstantTransformerBase(CodeTransformer):
+
+    def __init__(self, f):
+        self.f = f
+        super().__init__()
+
+    def visit_consts(self, consts):
+        return super().visit_consts(
+            tuple(
+                # This is all one expression.
+                frozenset(self.visit_consts(tuple(const)))
+                if isinstance(const, frozenset)
+                else self.visit_consts(const)
+                if isinstance(const, tuple)
+                else self.f(const)
+                if isinstance(const, self._type)
+                else const
+                ###
+                for const in consts
+            )
+        )
+
+
+def overloaded_constants(type_):
```

Comment:
I don't follow this?  We are dynamically creating subclasses of `_ConstantTransformerBase`, we're just doing it with a function instead of with a method of the base class.